apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -457,9 +469,21 @@ type Storage struct {
 	// may not be stable across reboots.
 	Name string `json:"name,omitempty"`
 
-	// Whether this disk represents rotational storage
+	// Whether this disk represents rotational storage.
+	// This field is not recommended for usage, please
+	// prefer using 'Type' field instead, this field
+	// will be deprecated eventually.
 	Rotational bool `json:"rotational,omitempty"`
 
+	// Device type, one of: HDD, SSD, NVME, UNKNOWN.
+	// When "Rotational" is "true", field value is set to "HDD".
+	// When "Rotational" is "false" and "Name" starts with "/dev/nvme",
+	// field value is set to "NVME".
+	// When previous cases are not covered, field value is set to "SSD".
+	// When "Name" undefined field value is set to "UNKNOWN".
```

Comment:
TBH I still don't think that all of these implementation details are a helpful thing to put in the CRD help. We just said the Rotational field was deprecated and to prefer this one, but then we define this one in terms of the Rotational field?

Reply:
I do not see any issues with removing all of comments that describe implementation details.
Will do on next rebase.

Reply:
done